api/api.go

```diff
@@ -615,8 +615,14 @@ func createNamespaceWithOwnerID(ctx context.Context, logger *logger.Logger, name
 
 	err = graphQLclient.Run(ctx, request, &response)
 
-	if err != nil {
-		return nil, errors.Wrapf(err, "Unable to create namespace %s for ownerId %s", name, ownerID)
+	// Only fall back to native graphql errors if there are no in-band errors.
+	if err != nil && len(response.Data.CreateNamespace.Errors) == 0 {
+		return nil, err
+	}
+
+	// Only fall back to out-of-band errors if there are no in-band errors
+	if len(response.Errors) > 0 && len(response.Data.CreateNamespace.Errors) == 0 {
+		return nil, response.Errors
```

Comment:
Having errors in 3 different place still seems like a very strange interface. A user having to be aware of errors from multiple places seems error prone. Why are there 2 different places to report errors in the graphQL response?

To reiterate my earlier argument, I think turning `response.Errors` into a typed `error` in `Run` would have the following advantages:
* no need to add `Errors` to every `Response` type (which is very error prone, and risks that errors are ignored)
* makes the API cleaner by consolidating all errors in a single place. Most callers will only need to check one variable (instead of 2-3). Any caller that does care about differentiating the error type can use  type checks

I can't think of any disadvantages to communicating all errors in the `error` response. Are there some I am missing?